src/views/HardwareInstallation.vue

```diff
@@ -396,7 +501,32 @@ export default {
 			},
 		},
 	},
+	watch: {
+		isLoading() {
+			console.info("Store data finally loaded");
+			return (this.hasData = true);
+		},
+	},
 	methods: {
+		reset(length) {
+			//reset this.deviceToAdd to undefined to prevent error from not updating model-value if manufacturer changed
+			if (length <= 1) {
+				this.deviceToAdd = this.selectManufacturer;
+			} else {
+				this.deviceToAdd = undefined;
+			}
+			return;
+		},
+		removeDuplicates(data) {
+			let res = data.filter(
+				(obj, index, self) =>
+					index ===
+					self.findIndex(
+						(t) => t.id === obj.id && t.text === obj.text,
+					),
+			);
+			return res;
+		},
```

Comment:
Bitte sprechende Variablennamen verwenden. Die Methode sieht eindeutig nach Copy/Paste aus z.B. Stack Overflow aus, was grundsätzlich als Quelle nicht falsch ist. Besser wäre gewesen, wenn in `getDeviceListOther` bereits darauf geachtet wird, keine Duplikate in die Liste aufzunehmen.

Reply:
Ja. Eine Standardfunktion. Wie könnte man es eleganter lösen? Ich habe da echt lange gesucht und ein ges ausprobiert. Das war für mich die einfachste und schnellste Lösung.

Reply:
Ok. Das mit den arrays war echt tricky. Für einen Ansatz wie man das unkompliziert lösen könnte, wäre ich sehr dankbar.